lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -4614,6 +4616,96 @@ LogicalResult ConvertAtenOp<AtenSqrtOp>::matchAndRewrite(
+        return rewriter.notifyMatchFailure(
+            op, "unimplemented: layout must be a constant");
+      else if (tensorLayout != torch_upstream::Layout::Strided)
+        return rewriter.notifyMatchFailure(
+            op, "unimplemented: layout is expected to be strided");
+    }
+    // Only `none`, `contiguous` and `preserve` memory_format are supported.
+    if (!op.getMemoryFormat().getType().template isa<Torch::NoneType>()) {
+      int64_t memoryFormat;
+      if (!matchPattern(op.getMemoryFormat(), m_TorchConstantInt(&memoryFormat)))
+        return rewriter.notifyMatchFailure(
+            op, "unimplemented: the memory format should be specified in "
+                "an integer constant");
+      if (memoryFormat != torch_upstream::MemoryFormat::Contiguous &&
+          memoryFormat != torch_upstream::MemoryFormat::Preserve)
+        return rewriter.notifyMatchFailure(
+            op, "unimplemented: only none, contiguous and preserve "
+                "memory_format is supported");
+    }
+
+    SmallVector<Value> size;
+    if (!getListConstructElements(op.getSize(), size))
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: size must be a ListConstruct");
+    SmallVector<Value> resultSize = getTypeConvertedValues(rewriter, loc, typeConverter,
+                                        size);
+    auto resultType =
+        typeConverter->convertType(op.getType()).template cast<RankedTensorType>();
+
+    DenseElementsAttr emptyVal;
+    if (op.getDtype().getType().template isa<Torch::NoneType>()) {
+      emptyVal  = DenseFPElementsAttr::get(resultType, {0.0F});
+    } else {
+      int64_t dtypeInt;
+      if (!matchPattern(op.getDtype(), m_TorchConstantInt(&dtypeInt)))
+        return rewriter.notifyMatchFailure(
+            op, "unimplemented: dtype must be a constant integer or none");
+      FailureOr<Type> maybeResultElementType = getTypeForScalarType(
+          op->getContext(), (torch_upstream::ScalarType)dtypeInt,
+          IntegerType::Signless);
+      if (failed(maybeResultElementType)) {
+        return rewriter.notifyMatchFailure(
+            op, "unable to convert `dtypeInt` to builtin type");
+      }
+      if(maybeResultElementType->isSignedInteger(64) || maybeResultElementType->isIndex())
+        emptyVal  = DenseIntElementsAttr::get(resultType, {0L});
+      if(maybeResultElementType->isSignedInteger(32))
+        emptyVal  = DenseIntElementsAttr::get(resultType, {0});
+      else if (maybeResultElementType->isSignlessInteger(64))
+        emptyVal  = DenseIntElementsAttr::get(resultType, {0UL});
+      else if (maybeResultElementType->isSignlessInteger(32))
+        emptyVal  = DenseIntElementsAttr::get(resultType, {0U});
+      else if (maybeResultElementType->isF64())
+        emptyVal  = DenseFPElementsAttr::get(resultType, {0.0});
+      else if (maybeResultElementType->isF32())
+        emptyVal  = DenseFPElementsAttr::get(resultType, {0.0F});
```

Comment:
Just for the sake of asking: did you test what happens if `maybeResultElementType` is a `bf16`?

Reply:
As I was expecting, it doesn't work (the whole reason for those `if`s is because there is a check in LLVM for the width of the element compared to the width of the type, and it doesn't cast the incoming type)

Reply:
I have yet to see the case in an app, so I'm thinking about leaving it out for the moment, since it's a bit more complex to have a bfloat `0.0` for this function

Reply:
The please add a final else branch that returns a rewriter.notifyMatchFailure when the type is not handled here.

Reply:
I tried to make it work with this method, but it makes it wide use of the C API and some functions that should not be called from the outside, so for the moment I'll just mark it as a `notifyMatchFailure` and if the case appears where we need to support bfloat16 here, I'll take a deeper look into making a proper implementation for it.